File: app/src/main/java/com/mklc/endlessrecyclerapp/data/network/repository/UserRepository.java
package com.mklc.endlessrecyclerapp.data.network.repository;

import com.mklc.endlessrecyclerapp.data.model.network.User;
import com.mklc.endlessrecyclerapp.data.network.ApiService;

import java.util.List;

import javax.inject.Inject;

import retrofit2.Call;

public class UserRepository {

    private final ApiService apiService;

    @Inject
    public UserRepository(ApiService apiService) {
        this.apiService = apiService;
    }

    public Call<List<User>> loadUsers(int page) {
        return apiService.getUsers(page);
    }
}

File: remotelib/src/main/java/com/mklc/remote/repository/UserRepository.kt
package com.mklc.remote.repository

import com.mklc.remote.api.ApiService
import com.mklc.remote.data.model.response.UserResponse
import io.reactivex.rxjava3.core.Single
import javax.inject.Inject

class UserRepository @Inject constructor(
    private val apiService: ApiService
) {

    fun loadUsers(): Single<List<UserResponse>> = apiService.loadUsers()

    fun loadUsers(page: Int): Single<List<UserResponse>> = apiService.loadUsers(page)
}